### ISSUE TITLE
Pasting text containing 'crlf' breaks section references

### ISSUE THREAD

**Issue by boltex** [leo-editor/leo-editor#3759]
See PR #3801.

**EKR**: This is definitely a bug.  Thanks for this clear description.

Doing this (pasting text containing **crlf** instead of **lf** line endings from outside editor)  yields 'orphan node' when writing external files that contain section references.

 errors caused because the putline method of findSectionName has problems with **crlf**. The putline process has problems detecting section references in body pane text if its line ending just before the << section ref >> signagure sould be detected.

Discovered this in leojs because it puts crlf by default on windows instead of linux. (until an external file is read from file.) So in leojs it was really annoying because it was happening mysteriously only in newly created outline parts, (parts read from disk of already existing files had only **lf** forced instead of **crlf**)  I fixed this by forcing text written in body pane to be eol **lf** instead of **crlf**.

I discovered it could happen in Leo if pasted from an external editor that would place text with crlf on the clipboard: I then tried to paste text (with **crlf** before section references in an outline of a tree) that I had in the clipboard in the real Leo, and the bug appeared there too! 

 Had to dig a bit to find out it was because of writing manualy in body pane in vscode (vscode puts **crlf** instead of **lf** unless forced otherwise.)

**Comment by tbpassin** [comment]
Is this coming from the section delimiter detection regex, do you think?  It contains a `\n`, which could well be the issue:

```
('section_delims', fr'^\s*{delim1}@@section-delims[ \t]+([^ \w\n\t]+)[ \t]+([^ \w\n\t]+)[ \t]*{delim2}$'),
```
I suppose this means that all input, whether from a file or a paste, should have its newlines normalized. Ouch, that could be a pain to track down.

**Comment by boltex** [comment]
Oops:  I had copied not directly into the body pane, which seems to handle new lines ok, I had pasted instead some 'outline content' with ctrl+shift+c from leojs, (not leo) which had crlf (because I did not fixed the new lines to ls only at that time) 
So this may totally be a non-issue. I'll make more tests and I'll come back with results soon to either pursue or close this issue.

**Comment by edreamleo** [comment]
Closed via PR #3801.
